src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.cs

```diff
@@ -1624,12 +1624,32 @@ public async Task GetAsync_UnicodeHostName_SuccessStatusCodeInResponse()
         {
             using (HttpClient client = CreateHttpClient())
             {
-                // international version of the Starbucks website
-                // punycode: xn--oy2b35ckwhba574atvuzkc.com
-                string server = "http://\uc2a4\ud0c0\ubc85\uc2a4\ucf54\ub9ac\uc544.com";
-                using (HttpResponseMessage response = await client.GetAsync(server))
+                try
                 {
-                    response.EnsureSuccessStatusCode();
+                    // international version of the Starbucks website
+                    // punycode: xn--oy2b35ckwhba574atvuzkc.com
+                    string server = "http://\uc2a4\ud0c0\ubc85\uc2a4\ucf54\ub9ac\uc544.com";
+                    using (HttpResponseMessage response = await client.GetAsync(server))
+                    {
+                        response.EnsureSuccessStatusCode();
+                    }
+                }
+                catch (Exception ex)
+                {
+                    try
+                    {
+                        // a simple WebApp running on Azure
+                        // punycode: xn--nicode-2ya.azurewebsites.net
+                        string backupServer = "http://�nicode.azurewebsites.net/time";
```

Comment:
We usually don't have explicit unicode in our source files. So, you should use "\uXXXX" notation.

Reply:
Changed. Is there a reason why we do so in cases where the characters have normal symbols? In this case it adds an extra step for anyone trying to look at the address manually.

Reply:
>Is there a reason why we do so in cases where the characters have normal symbols?

By "normal", do you mean "ASCII"?  Historically, many of our command line tools don't do a good job handling text files with unicode characters. So, as a development pattern, we tend to avoid putting non-ASCII characters into source code.

@stephentoub

Reply:
> By "normal", do you mean "ASCII"

I was thinking of those that are displayed nicely by the browser (i.e. not turned into ?).

>many of our command line tools don't do a good job

I just looked at Github Desktop and it also fails to render `ü` and instead goes for �. Looks like I was too optimistic about Unicode support :/